File: jtaf-web/src/main/java/ch/jtaf/control/report/CompetitionCsvExport.java
package ch.jtaf.control.report;

import ch.jtaf.entity.Athlete;
import ch.jtaf.data.CompetitionRankingCategoryData;
import ch.jtaf.data.CompetitionRankingData;

public class CompetitionCsvExport {

    private final CompetitionRankingData ranking;

    public CompetitionCsvExport(CompetitionRankingData ranking) {
        this.ranking = ranking;
    }

    public String create() {
        StringBuilder sb = new StringBuilder();
        for (CompetitionRankingCategoryData category : ranking.getCategories()) {
            int position = 1;
            for (Athlete athlete : category.getAthletes()) {
                sb.append(position);
                sb.append(",");
                sb.append(athlete.getLastName());
                sb.append(",");
                sb.append(athlete.getFirstName());
                sb.append(",");
                sb.append(category.getCategory().getAbbreviation());
                sb.append("\n");
                position++;
            }
        }
        return sb.toString();
    }

}

File: src/main/kotlin/ch/jtaf/control/reporting/report/CompetitionCsvExport.kt
package ch.jtaf.control.reporting.report

import ch.jtaf.control.reporting.data.CompetitionRankingData

class CompetitionCsvExport(private val ranking: CompetitionRankingData) {

    fun create(): String {
        val sb = StringBuilder()
        for (category in ranking.categories) {
            var position = 1
            for (athlete in category.athletes) {
                sb.append(position)
                sb.append(",")
                sb.append(athlete.athlete.lastName)
                sb.append(",")
                sb.append(athlete.athlete.firstName)
                sb.append(",")
                sb.append(category.category.abbreviation)
                sb.append("\n")
                position++
            }
        }
        return sb.toString()
    }

}
